prov/sockets/src/sock_progress.c

```diff
@@ -169,6 +172,13 @@ static void sock_pe_release_entry(struct sock_pe *pe,
 		return;
 	}
 
+	if (pe_entry->type == SOCK_PE_TX) {
+		/* Reinit TX communication ring bufers */
+		pe_entry->comm_buf.rcnt =
+			pe_entry->comm_buf.wcnt =
+			pe_entry->comm_buf.wpos = 0;
+	}
```

Comment:
Please add a new ring buffer call for this, ofi_rbreset, or call ofi_rbdiscard(rb, ofi_rbused(rb))